### PR TITLE
Enhance Android & iOS documentation

### PR DESCRIPTION
**Pull request by saibotma**
I noticed that the documentation of conference events was outdated or contained wrong information in many cases.
Someone please needs to check that all the data sent in conference events are correctly documented, as I did not take a look at the React code (I assume) that creates those events and their data.

If you can give me a hint where I find the peace of code that constructs the events, then I can check it by myself.

### CODE REVIEW

File: docs/dev-guide/android-sdk.md [hunk 1]
```diff
@@ -370,68 +371,90 @@ Please see `JitsiMeetActivity`, which registers for all the events and can serve
...
-Depending of whether the participant is the local one or not, some of them are 
-present/missing.
-    isLocal
-    email
-    name
-    participantId
+
+Broadcasted when a participant has joined the conference. `data` contains the following information:
+
+- `email`: the email of the participant.
```
Comment (saibotma) [inline docs/dev-guide/android-sdk.md:403]: In my example app, the `email` key never existed. Should this be removed?

File: docs/dev-guide/android-sdk.md [hunk 2]
```diff
@@ -370,68 +371,90 @@ Please see `JitsiMeetActivity`, which registers for all the events and can serve
...
-    isLocal
-    email
-    name
-    participantId
+
+Called when a participant has left the conference. `data` contains the following information:
+
+- `participantId`: the id of the participant that left.
 
 ##### ENDPOINT_TEXT_MESSAGE_RECEIVED
```
Comment (saibotma) [inline docs/dev-guide/android-sdk.md:414]: What actually is the difference between an “endpoint text message” and a “chat message”?

File: docs/dev-guide/android-sdk.md [hunk 3]
```diff
@@ -370,68 +371,90 @@ Please see `JitsiMeetActivity`, which registers for all the events and can serve
...
-Broadcasted when a RETRIEVE_PARTICIPANTS_INFO action is called.
-The `data` HashMap contains a `participantsInfo` key with a list of participants information and a `requestId` key with the id that was sent in the RETRIEVE_PARTICIPANTS_INFO action.
+
+Broadcasted when a RETRIEVE_PARTICIPANTS_INFO action is called. The `data` HashMap contains a `participantsInfo` key
+with a list of participants information and a `requestId` key with the id that was sent in the
+RETRIEVE_PARTICIPANTS_INFO action.
 
 ##### CHAT_MESSAGE_RECEIVED
-Broadcasted when a chat text message is received.
-The `data` HashMap contains a `senderId` key with the participantId of the sender, a `message` key with the content, a `isPrivate` key with a boolean value and a `timestamp` key.
```
Comment (saibotma) [inline docs/dev-guide/android-sdk.md:425]: The `timestamp` key never existed in the example I tried. Thus, I removed it.

File: docs/dev-guide/android-sdk.md (hunk 1 above)
Comment (saghul) [reply docs/dev-guide/android-sdk.md:403]: It may not be set if the remote participant didn't set one.

File: docs/dev-guide/android-sdk.md (hunk 2 above)
Comment (saghul) [reply docs/dev-guide/android-sdk.md:414]: It's subtle indeed. Chat messages go through the chat channel, which is designed for text based communication between humans, and endpoint messages allow SDK users to exchange arbitrary data over a connection which follows the media path.

File: docs/dev-guide/android-sdk.md (hunk 3 above)
Comment (saghul) [reply docs/dev-guide/android-sdk.md:425]: It's optional, it may or may not be there, please leave it there.

**Review by saghul — CHANGES_REQUESTED** [review]
Thank you, this is very helpful! I left a couple of comments.

**Comment by saibotma** [comment]
Thanks for the information. You can take another look.

**Review by saghul — APPROVED** [review]

**Comment by saghul** [comment]
Thank you!

PS: very cool use of Jitsi in your app!

**Pull request merged**